app/upload/models/audio.model.ts

```diff
@@ -0,0 +1,153 @@
+import {Subscription} from 'rxjs';
+import {Upload} from '../services/upload.service';
+import {HalDoc} from '../../shared/cms/haldoc';
+
+export class AudioModel {
+
+  static STAT_UPLOADING = ['uploading', 'upload failed', 'saving', 'save failed'];
+
+  static STAT_PROCESSING = ['uploaded', 'validating', 'valid', 'invalid',
+    'complete', 'creating mp3s', 'creating mp3s failed', 'mp3s created'];
+
+  static STAT_DONE = ['done'];
+
+  static STAT_ERROR = ['upload failed', 'save failed', 'invalid', 'creating mp3s failed'];
+
+  // remote properties
+  id: number;
+  filename: string;
+  label: string;
+  size: number;
+  duration: number;
+  position: number;
+  status: string;
+
+  // state of the upload
+  progress: number;
+  progressSub: Subscription;
+
+  // existing docs
+  doc: HalDoc;
+
+  // new uploads
+  upload: Upload;
+  version: HalDoc;
+
+  constructor(data: any = {}) {
+    let flds = ['id', 'filename', 'label', 'size', 'duration', 'position', 'status'];
+    for (let key of Object.keys(data)) {
+      if (flds.indexOf(key) > -1) {
+        this[key] = data[key];
+      }
+    }
+  }
+
+  static fromDoc(doc: HalDoc): AudioModel {
+    // TODO: smoke and mirrors
+    doc['status'] = (doc['status'] === 'uploaded') ? 'done' : doc['status'];
```

Comment:
For now, I just mark any uploaded file as `done`, so the progress bar disappears.  But eventually (when CMS takes the file through all those `STAT_PROCESSING`) this can be removed.